pyasic/web/vnish.py

```diff
@@ -143,3 +150,16 @@ async def autotune_presets(self) -> dict:
 
     async def find_miner(self) -> dict:
         return await self.send_command("find-miner", privileged=True)
+
+    async def update_firmware(self, file: Path, keep_settings: bool = True) -> dict:
+        """Perform a system update by uploading a firmware file and sending a command to initiate the update."""
+        async with aiofiles.open(file, "rb") as firmware:
+            file_content = await firmware.read()
+        parameters = {
+            "file": (file.name, file_content, "application/octet-stream"),
+            "keep_settings": keep_settings
+        }
+        return await self.send_command(
+            command="system/upgrade",
+            **parameters
+        )
```

Comment:
This doesn't seem correct, it seems like they have to be constructed in a weird way, here's a source from BTCTools - https://github.com/btccom/libbtctools/blob/da6e836249fd70010bf854a647e7e2096424782a/src/lua/scripts/upgrader/AnthillOS.lua#L58-L78

Reply:
I tried to refactor the code according to the reference `update_firmware`. Please take a look